### PR TITLE
Improving Pipelines by defaulting to framework='tf' when pytorch seems unavailable.

### PR DESCRIPTION
**Pull request by Narsil**
# What does this PR do?

When loading a model that was `tf` only and by passing only model by string without framework argument, it would fail with an odd error message:

```python
>>> transformers.AutoModel.from_pretrained('Narsil/small')
OSError: Can't load weights for 'Narsil/small'. Make sure that:
- 'Narsil/small' is a correct model identifier listed on 'https://huggingface.co/models' (It exists and contains tf_model.h5)
- or 'Narsil/small' is the correct path to a directory containing a file named one of pytorch_model.bin, tf_model.h5, model.ckpt.
```

This PR corrects the `get_framework` that happens very early in the pipeline to detect the type of model
automatically. It does trigger an early download, but that will happen anyway later. 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dimiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests), 
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to the it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [x] Did you write any new necessary tests? 

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors which may be interested in your PR.

@mfuntowicz

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
LGTM, thanks @Narsil!

**Comment by Narsil** [comment]
Actually, This introduced a pretty big bug where

```python
nlp = pipeline(task) 
```

Would not work anymore. I tried a different solution around that.

File: tests/test_pipelines.py [hunk 1]
```diff
@@ -823,3 +833,4 @@ def test_pt_defaults(self):
         for task in self.pipelines:
             with self.subTest(msg="Testing Torch defaults with PyTorch and {}".format(task)):
                 pipeline(task, framework="pt")
+                pipeline(task)
```
Comment (LysandreJik) [inline tests/test_pipelines.py:836]: Thanks for completing the suite!

**Review by LysandreJik — APPROVED** [review]
Cool, great that you completed the CI.

**Pull request merged**